### ISSUE TITLE
Summary splits sentences just by dots - but not by :?!

### ISSUE THREAD

**Issue by lerni** [silverstripe/silverstripe-framework#10167]
## Affected Version
4.x

## Description
DBText::Summary() splits sentences, which is nice, but it looks as it doesn't respect question & exclamation marks or colons, which it should?

https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/FieldType/DBText.php#L133-L137

Wolud a fix like bellow (L136) be acceptable?
```
preg_split('@(?<=[.?!:])@', $value)
instead of
preg_split('@(?<=\.)@', $value)
```

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/10181
- https://github.com/silverstripe/silverstripe-framework/pull/11100
- https://github.com/silverstripe/developer-docs/pull/428

**Comment by michalkleiner** [comment]
I'd accept a fix where the set of characters for sentence splitting were managed via a static config with the default of the single full stop, as is now, and then from next minor version we could enhance the set of characters to what you suggest, while others could use the config earlier, if they wanted.

**Comment by lerni** [comment]
@michalkleiner something like that you had in mind?
https://github.com/lerni/silverstripe-framework/commit/c1f0e1280570dbe1c69a4c5eac24599785d204cb

**Comment by emteknetnz** [comment]
Because it has new config it would need to be the `4` branch. It would be released in 4.11.0

**Comment by michalkleiner** [comment]
And since it's now targetting `4` and will be part of `4.11`, we can mention it in the changelog and add the extra characters into the new config.

**Comment by lerni** [comment]
https://github.com/silverstripe/silverstripe-framework/pull/11100

**Comment by GuySartorelli** [comment]
Reopening as the prs have not been merged yet

**Comment by GuySartorelli** [comment]
PRs merged - will be released in CMS 5.2 in April
